src/parser/vast_parser.js

```diff
@@ -125,254 +142,279 @@ export class VASTParser extends EventEmitter {
    * @emits    VASTParserr#VAST-resolved
    * @callback cb
    */
-  getAndParse(url, options, cb) {
+  getAndParseVAST(url, options, cb) {
     if (!cb) {
       if (typeof options === 'function') {
         cb = options;
         options = {};
       } else {
-        throw new Error(
-          'VASTParser getAndParse method called without valid callback function'
-        );
+        throw new Error('Method called without valid callback function');
       }
     }
 
-    this.parentURLs = [];
+    this.initParsingStatus(options);
 
-    this.fetchVAST(url, options)
+    this.fetchVAST(url)
       .then(xml => {
         options.originalUrl = url;
-        this.parse(xml, options, cb);
+        this.parse(xml, options)
+          .then(ads => {
+            const response = new VASTResponse();
+            response.ads = ads;
+            response.errorURLTemplates = this.errorURLTemplates;
+            this.completeWrapperResolving(response);
+
+            cb(null, response);
+          })
+          .catch(err => cb(err));
       })
       .catch(err => cb(err));
   }
 
   /**
    * Parses the given xml Object into a VASTResponse.
    * Executes the callback with either an error or the fully parsed VASTResponse.
-   * @param    {Object} vastXml - An object representing an xml document.
+   * @param    {Object} vastXml - An object representing a vast xml document.
    * @param    {Object} options - An optional Object of parameters to be used in the parsing process.
    * @emits    VASTParserr#VAST-resolving
    * @emits    VASTParserr#VAST-resolved
    * @callback cb
    */
-  parse(vastXml, options, cb) {
+  parseVAST(vastXml, options, cb) {
```

Comment:
Looks like slightly different of `getAndParseVAST()` because it does not fetch the URL. But where is it used?

Reply:
`getAndParseVAST` and `parseVAST` are the 2 methods we were already exposing as public API of the parser. I just kept it in order to provide an option to parse an already fetched VAST.